### PR TITLE
Remove connector allow-all access control

### PR DESCRIPTION
**Pull request by kokosing**
Remove connector allow-all access control

System connector access control does exactly the same and does not break
the system roles. Notice that allow-all connector access control says to
the engine that it controls connector roles, while it has no roles.

### CODE REVIEW

File: core/trino-main/src/test/java/io/trino/connector/AllowAllAccessControl.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.trino.plugin.base.security;
+package io.trino.connector;
```
Comment (kokosing) [inline core/trino-main/src/test/java/io/trino/connector/AllowAllAccessControl.java:14]: It is is used only by tests now, by mock connector access control.

Comment (findepi) [reply core/trino-main/src/test/java/io/trino/connector/AllowAllAccessControl.java:14]: what about renaming the class to eg TestingAllowAllAccessControl ?

**Review by findepi — APPROVED** [review]

**Comment by kokosing** [comment]
Addressed comments (AC).

**Comment by kokosing** [comment]
I cannot remove it :(

It appeared that `allow-all` not only controls access but also controls the ownership. So if a you set system access control then ownership of tables and views is controlled by system metadata.

**Pull request closed without merging**